ansible/roles/fanout/tasks/fanout_sonic.yml

```diff
@@ -31,7 +34,17 @@
     sonic/fanout_sonic_202012.yml
   when: "'20201231' in fanout_sonic_version['build_version']"
 
-- name: deploy SONiC fanout with image version 202205
+- name: deploy SONiC fanout with image version 202205 not incremental and not dry_run
   include_tasks:
     sonic/fanout_sonic_202205.yml
-  when: "'20220531' in fanout_sonic_version['build_version']"
+  when: "'20220531' in fanout_sonic_version['build_version'] and dry_run is not defined and incremental is not defined"
+
+- name: deploy SONiC fanout with image version 202205 incremental and not dry_run
+  include_tasks:
+    sonic/fanout_sonic_incremental_202205.yml
+  when: "'20220531' in fanout_sonic_version['build_version'] and dry_run is not defined and incremental is defined"
+
+- name: deploy SONiC fanout with image version 202205 dry_run
+  include_tasks:
+    sonic/fanout_sonic_dry_run_202205.yml
+  when: "'20220531' in fanout_sonic_version['build_version'] and dry_run is defined"
```

Comment:
I think put line 37 to 50 in a block can make it more clear:

```
- name: deploy SONiC fanout with image version 202012
  block:
    - name: step 1
      ...
    - name: step 2
      ...
  when: "'20220531' in fanout_sonic_version['build_version']"
```

Reply:
Updated